src/server/lib/encryptedStateCookie.ts

```diff
@@ -35,9 +35,17 @@ export const setEncryptedStateCookie = (
 export const readEncryptedStateCookie = (
   req: Request,
 ): EncryptedState | undefined => {
-  const encryptedCookie = baseUri.includes('localhost')
-    ? req.cookies[encryptedStateCookieName]
-    : req.signedCookies[encryptedStateCookieName];
+  // TODO: We need to double check the security implications of using an env variable
+  // to read from the unsigned cookie storage.
+  const encryptedCookie =
+    baseUri.includes('localhost') || process.env.RUNNING_IN_CYPRESS === 'true'
```

Comment:
```suggestion
    process.env.RUNNING_IN_CYPRESS === 'true'
```

I don't think we need the `localhost` check here, as we never want to run gateway on localhost

Reply:
ok we can remove that, but it isn't being added by this PR.
What about [encryptedStateCookieOptions](https://github.com/guardian/gateway/blob/4c574d435a5eef78fc6528547b9050ed6bc4289b/src/server/lib/encryptedStateCookie.ts#L13-L14) in this same file? it's also checking that baseUri includes localhost there.. 
Would this logic have to match the encryptedStateCookieOptions ? 
I'm a bit confused because otherwise this could have checked  `encryptedStateCookieOptions.secure` or `encryptedStateCookieOptions.signed` instead of look at the baseUri directly, right ?

Reply:
I agree, the `localhost` check there doesn't seem to add any information. I think we should keep [the check in `encryptedStateCookieOptions`](https://github.com/guardian/gateway/blob/7f805dd1a982c98d8557be182edf6cd65391cfbc/src/server/lib/encryptedStateCookie.ts#L11-L16) though - don't Cypress mocked tests run Gateway on `localhost`? I suppose it also prevents the possibility of trying to set a secure cookie on an insecure connection. Then again, from the [cookie module docs](https://www.npmjs.com/package/cookies#readme):

> An error will be thrown if you try to send secure cookies over an insecure socket.